examples/radial_histogram_equalization.py

```diff
@@ -0,0 +1,119 @@
+"""
+=============================
+Radial Histogram Equalization
+=============================
+
+This example applies the Radial Histogram Equalizing Filter (`sunkit_image.radial.rhef`) to a sunpy map.
+"""
+
+import astropy.units as u
+import matplotlib.pyplot as plt
+import sunpy.data.sample
+import sunpy.map
+
+import sunkit_image.enhance as enhance
+import sunkit_image.radial as radial
+from sunkit_image.utils import equally_spaced_bins
+
+#######################################################################################
+# Let us use the sunpy sample data AIA image to showcase the RHE filter.
+
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+
+# Create radial segments (RHEF should use a dense grid)
+radial_bin_edges = equally_spaced_bins(0, 2, aia_map.data.shape[0] // 2)
+radial_bin_edges *= u.R_sun
+
+rhef_map = radial.rhef(aia_map, radial_bin_edges, do_vignette=True, do_progressbar=True)
+
+
+# Plot the 2 maps in a single figure with one row and three columns
+fig, axes = plt.subplots(1, 2, figsize=(10, 5), sharex="all", sharey="all", subplot_kw={"projection": aia_map})
+
+# Original AIA map.plot()
+ax = axes[0]
+aia_map.plot(axes=ax, clip_interval=(1, 99.99) * u.percent)
+ax.set_title("Original AIA Map")
+
+# RHEF map.plot()
+ax = axes[1]
+rhef_map.plot(axes=ax, norm=None)  # This norm=None is very important for visualization
+ax.set_title("RHE Filtered Map")
+
+plt.tight_layout()
+plt.show()
+
+
+#######################################################################################
+# The RHEF has one free parameter that works in post processing to modulate the output.
+# Here are some of the choices one could make.
+# See the thesis (Gilly 2022) Eq 4.15 for details about upsilon: https://www.proquest.com/docview/2759080511
+
+# Define the list of upsilon pairs where the first number affects dark components and the second number affects bright ones
+upsilon_list = [
+    0.35,
+    None,
+    (0.1, 0.1),
+    (0.5, 0.5),
+    (0.8, 0.8),
+]
+
+# Create a figure with subplots for each upsilon pair plus the original map
+fig, axes = plt.subplots(2, 3, figsize=(15, 10), sharex="all", sharey="all", subplot_kw={"projection": aia_map})
+axs = axes.flatten()
+
+aia_map.plot(axes=axs[0], clip_interval=(1, 99.99) * u.percent)
+axs[0].set_title("Original AIA Map")
+
+# # Loop through the upsilon_list and plot each filtered map
+for i, upsilon in enumerate(upsilon_list):
+    out_map = radial.rhef(aia_map, upsilon=upsilon, do_vignette=True, method="scipy")
+    out_map.plot(axes=axs[i + 1], norm=None)
+    axs[i + 1].set_title(f"Upsilon = {upsilon}")
+
+# Adjust layout
+plt.tight_layout()
+plt.show()
```

Comment:
```suggestion
```

Reply:
I was wrong before, I think that the tight_layout does make an improvement

Reply:
Fair enough, but the `plt.show` has to still go.
Sometime ago, we decided that gallery examples should run all the way to the end before showing any plots.